command/watch.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/gotify/go-api-client/v2/models"
+	"gopkg.in/urfave/cli.v1"
+)
+
+func Watch() cli.Command {
+	return cli.Command{
+		Name:      "watch",
+		Usage:     "watch the result of a command and pushes output difference",
+		ArgsUsage: "<cmd>",
+		Flags: []cli.Flag{
+			cli.Float64Flag{Name: "interval,n", Usage: "watch interval (sec)", Value: 2.},
+			cli.IntFlag{Name: "priority,p", Usage: "Set the priority"},
+			cli.StringFlag{Name: "exec,x", Usage: "Pass command to exec (default to \"sh -c\")", Value: "sh -c"},
+			cli.StringFlag{Name: "title,t", Usage: "Set the title (empty for command)"},
+			cli.StringFlag{Name: "token", Usage: "Override the app token"},
+			cli.StringFlag{Name: "url", Usage: "Override the Gotify URL"},
+		},
+		Action: doWatch,
+	}
+}
+
+func doWatch(ctx *cli.Context) {
+	conf, confErr := config.ReadConfig(config.GetLocations())
+
+	cmdArgs := ctx.Args()
+	cmdStringNotation := strings.Join(cmdArgs, " ")
+	execArgs := strings.Split(ctx.String("exec"), " ")
+	cmdArgs = append(execArgs[1:], cmdStringNotation)
+	execCmd := execArgs[0]
+
+	interval := ctx.Float64("interval")
+	priority := ctx.Int("priority")
+	title := ctx.String("title")
+	if title == "" {
+		title = cmdStringNotation
+	}
+	token := ctx.String("token")
+	if token == "" {
+		if confErr != nil {
+			utils.Exit1With("token is not configured, run 'gotify init'")
+			return
+		}
+		token = conf.Token
+	}
+	stringURL := ctx.String("url")
+	if stringURL == "" {
+		if confErr != nil {
+			utils.Exit1With("url is not configured, run 'gotify init'")
+			return
+		}
+		stringURL = conf.URL
+	}
+	parsedURL, err := url.Parse(stringURL)
+	if err != nil {
+		utils.Exit1With("invalid url", stringURL)
+		return
+	}
+
+	watchInterval := time.Duration(interval*1000) * time.Millisecond
+
+	evalCmdOutput := func() (string, error) {
+		cmd := exec.Command(execCmd, cmdArgs...)
+		timeOut := time.After(watchInterval)
+		outputBuf := bytes.NewBuffer([]byte{})
+		cmd.Stdout = outputBuf
+		cmd.Stderr = outputBuf
+		err := cmd.Start()
+		if err != nil {
+			return "", fmt.Errorf("command failed to invoke: %v", err)
+		}
+		done := make(chan error)
+		go func() {
+			err := cmd.Wait()
+			if err != nil {
+				done <- fmt.Errorf("command failed to invoke: %v", err)
+			}
+			done <- nil
+		}()
+		select {
+		case err := <-done:
+			return outputBuf.String(), err
+		case <-timeOut:
+			cmd.Process.Kill()
+			return outputBuf.String(), errors.New("command timed out")
+		}
+	}
+
+	lastOutput, err := evalCmdOutput()
+	if err != nil {
+		utils.Exit1With("first run failed", err)
+	}
+	for execTime := range time.NewTicker(watchInterval).C {
+		output, err := evalCmdOutput()
+		if err != nil {
+			output += fmt.Sprintf("\n!== <%v> ==!", err)
+		}
+		if output != lastOutput {
+			msgData := bytes.NewBuffer([]byte{})
+			fmt.Fprintf(msgData, "command output for \"%s\" changed at %v\n\n", cmdStringNotation, execTime)
+			fmt.Fprintln(msgData, "== BEGIN OLD OUTPUT ==")
```

Comment:
I think an option for omiting this info would be useful.
Imagine a command like this:
```
./cli watch "http https://gotify.net --check-status &> /dev/null && echo gotify.net is online || echo gotify.net is offline"
```
Only the new value is relevant and therefore should only be present in the message.

Reply:
Okay.